pkg/specgen/generate/namespaces.go

```diff
@@ -247,6 +264,22 @@ func namespaceOptions(s *specgen.SpecGenerator, rt *libpod.Runtime, pod *libpod.
 			return nil, fmt.Errorf("looking up container to share user namespace with: %w", err)
 		}
 		toReturn = append(toReturn, libpod.WithUserNSFrom(userCtr))
+	case specgen.Private:
+		if err := userNSConflictsWithPod(pod, s.UserNS.NSMode); err != nil {
+			return nil, err
+		}
+	case specgen.Auto:
+		if err := userNSConflictsWithPod(pod, s.UserNS.NSMode); err != nil {
+			return nil, err
+		}
+	case specgen.NoMap:
+		if err := userNSConflictsWithPod(pod, s.UserNS.NSMode); err != nil {
+			return nil, err
+		}
+	case specgen.Path:
+		if err := userNSConflictsWithPod(pod, s.UserNS.NSMode); err != nil {
+			return nil, err
+		}
```

Comment:
That is a bit inelegant and does not scale well. IT could be made much simpler by moving this out of the switch case.

since you already check the userns mode != FromPod in userNSConflictsWithPod() all you need to do is call `userNSConflictsWithPod(pod,s.UserNS.NSMode)` once before or after this switch case block.